### ISSUE TITLE
"Argument #2 ($properties) must be of type array, null given" thrown on Snaptcha pages

### ISSUE THREAD

**Issue by carolinecblaker** [putyourlightson/craft-snaptcha#24]
I just upgraded my website to Craft CMS 4 and pages with Snaptcha are throwing an error via Typecast. It looks like the $config is null. I don't know if this is a bug or not.

I have tried:
- resaving the config multiple times.

`craft\helpers\Typecast::properties(): Argument #2 ($properties) must be of type array, null given, called in /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/base/Model.php`

`TypeError: craft\helpers\Typecast::properties(): Argument #2 ($properties) must be of type array, null given, called in /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/base/Model.php on line 69 and defined in /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/helpers/Typecast.php:44
Stack trace:
#0 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/base/Model.php(69): craft\helpers\Typecast::properties('putyourlightson...', NULL)
#1 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/putyourlightson/craft-snaptcha/src/variables/SnaptchaVariable.php(30): craft\base\Model->__construct(NULL)
#2 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Extension/CoreExtension.php(1607): putyourlightson\snaptcha\variables\SnaptchaVariable->getFieldValue()
#3 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/helpers/Template.php(101): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(putyourlightson\snaptcha\variables\SnaptchaVariable), 'fieldValue', Array, 'any', false, false)
#4 /srv/users/petroglyph-2021/apps/petroglyph-2021/storage/runtime/compiled_templates/7c/7c82eb5389eecf938bc7bb6277e86e28.php(117): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(putyourlightson\snaptcha\variables\SnaptchaVariable), 'fieldValue', Array)
#5 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(394): __TwigTemplate_fc2e4f5f93c4f3ad10d6a5a73011ac8b->doDisplay(Array, Array)
#6 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#7 /srv/users/petroglyph-2021/apps/petroglyph-2021/storage/runtime/compiled_templates/c0/c05edea58a10e5160c66227cdc119252.php(88): Twig\Template->display(Array)
#8 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(171): __TwigTemplate_d2f329751bba632e81419e7664197185->block_content(Array, Array)
#9 /srv/users/petroglyph-2021/apps/petroglyph-2021/storage/runtime/compiled_templates/e9/e936ca7b28cc85922cfa9daa06224f2b.php(115): Twig\Template->displayBlock('content', Array, Array)
#10 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(394): __TwigTemplate_dc978274b1ce1422271396de7c4558a7->doDisplay(Array, Array)
#11 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#12 /srv/users/petroglyph-2021/apps/petroglyph-2021/storage/runtime/compiled_templates/c0/c05edea58a10e5160c66227cdc119252.php(46): Twig\Template->display(Array, Array)
#13 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(394): __TwigTemplate_d2f329751bba632e81419e7664197185->doDisplay(Array, Array)
#14 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#15 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#16 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#17 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#18 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/web/View.php(428): Twig\Environment->render('/contact.html', Array)
#19 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/web/View.php(481): craft\web\View->renderTemplate('/contact.html', Array)
#20 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(56): craft\web\View->renderPageTemplate('/contact.html', Array, 'site')
#21 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\TemplateResponseFormatter->format(Object(craft\web\Response))
#22 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#23 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#24 /srv/users/petroglyph-2021/apps/petroglyph-2021/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#25 /srv/users/petroglyph-2021/apps/petroglyph-2021/public/index.php(21): yii\base\Application->run()
#26 {main}
`

**Comment by carolinecblaker** [comment]
<img width="1081" alt="Screen Shot 2022-08-25 at 3 11 03 PM" src="https://user-images.githubusercontent.com/238110/186769850-236c0209-f747-48f4-b176-d8f152cce4eb.png">
Here's a SS of the formatted stack in case it's helpful.

**Comment by bencroker** [comment]
> I have tried: resaving the config multiple times.

Can you explain what you mean by "resaving the config"? How exactly are you calling the Snaptcha field, or the field value?

**Comment by carolinecblaker** [comment]
> Can you explain what you mean by "resaving the config"? 

In keeping with debugging techniques that have worked in other scenarios, I've gone into the settings page and resaved it several times- also tried changing some settings and hitting save just to see if it would make the error go away.

**Comment by carolinecblaker** [comment]
> How exactly are you calling the Snaptcha field, or the field value?

I was using separate name and value tags like:

```
<input type="hidden" id="my-snaptcha-field"
    name="{{ craft.snaptcha.fieldName }}" 
    value="{{ craft.snaptcha.fieldValue }}">
```

**Comment by bencroker** [comment]
It looks like this was a bug. Fixed in https://github.com/putyourlightson/craft-snaptcha/commit/5af964f9aef63c6d1fe270c4db3d4465ffebfdb2 and released in version 4.0.1.
